src/nnue/nnue_misc.cpp

```diff
@@ -104,7 +106,8 @@ void format_cp_aligned_dot(Value v, std::stringstream& stream, const Position& p
 
 // Returns a string with the value of each piece on a board,
 // and a table for (PSQT, Layers) values bucket by bucket.
-std::string trace(Position& pos, const Eval::NNUE::Networks& networks) {
+std::string
+trace(Position& pos, const Eval::NNUE::Networks& networks, Eval::NNUE::AccumulatorCaches& caches) {
```

Comment:
can we now just pass a nullptr inside trace for the bignet cache?

Reply:
not really, 
see https://github.com/official-stockfish/Stockfish/pull/5183/files#diff-231ff7ea9252f67f0a5cbef26e4b2371d91f9f7558fd417d50dbada4247a9ad6R807,
we still need a valid pointer here. I've added an assert to `update_accumulator_refresh_cache` for the future and we got another comment here pointing that out https://github.com/official-stockfish/Stockfish/pull/5183/files#diff-5e10e0eae2cec9399c889f5f37b1ed2f8c0ec975f2c7dd808f127df808b27becR98

we could determine the path to take depending on if the pointer is a nullptr but the constexpr if doesnt cost us anything during runtime